tmail-backend/jmap/extensions/src/main/java/com/linagora/tmail/james/jmap/StandaloneEventAttendanceRepository.java

```diff
@@ -0,0 +1,129 @@
+package com.linagora.tmail.james.jmap;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+
+import jakarta.inject.Inject;
+import jakarta.mail.Flags;
+
+import org.apache.james.core.Username;
+import org.apache.james.mailbox.MailboxSession;
+import org.apache.james.mailbox.MessageIdManager;
+import org.apache.james.mailbox.MessageManager;
+import org.apache.james.mailbox.SessionProvider;
+import org.apache.james.mailbox.model.FetchGroup;
+import org.apache.james.mailbox.model.MessageId;
+import org.apache.james.mailbox.model.MessageResult;
+import org.reactivestreams.Publisher;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import reactor.core.publisher.Mono;
+
+public class StandaloneEventAttendanceRepository implements EventAttendanceRepository {
+    private static final Logger LOGGER = LoggerFactory.getLogger(StandaloneEventAttendanceRepository.class);
+    private static final List<String> EVENT_ATTENDANCE_FLAGS = List.of("$accepted", "$rejected", "$tentativelyaccepted", "$needs-action");
+
+    private final MessageIdManager messageIdManager;
+    private final SessionProvider sessionProvider;
+
+    @Inject
+    public StandaloneEventAttendanceRepository(MessageIdManager messageIdManager, SessionProvider sessionProvider) {
+        this.messageIdManager = messageIdManager;
+        this.sessionProvider = sessionProvider;
+    }
+
+    @Override
+    public Publisher<AttendanceStatus> getAttendanceStatus(Username username, MessageId messageId) {
+        MailboxSession systemMailboxSession = sessionProvider.createSystemSession(username);
+
+        return getFlags(messageId, systemMailboxSession)
+            .flatMap(this::getAttendanceStatusFromFlags)
+            .switchIfEmpty(handleMissingEventAttendanceFlag(messageId, systemMailboxSession));
+    }
+
+    private Mono<AttendanceStatus> handleMissingEventAttendanceFlag(MessageId messageId, MailboxSession systemMailboxSession) {
+        return Mono.from(getFlags(messageId, systemMailboxSession)).map(flags -> {
+            LOGGER.warn("No event attendance flag found for message {}", messageId);
+            LOGGER.warn("Flags for message {}: {}", messageId, flags);
+            LOGGER.warn("Defaulting to NeedsAction for message {}", messageId);
```

Comment:
BTW what we need getlag for? If it's just add it into the debug log it is not worth the extra DB read...

Reply:
This log will be triggered when no attendance flags exist for that message. We log the flags in case an attendance flag was attached to the message, but we failed to recognize that.

Reply:
Downgraded the log level to debug.